### ISSUE TITLE
Update rule S1751: cover all and only cases when loop is executed once

### ISSUE THREAD

**Issue by vilchik-elena** [SonarSource/SonarJS#955]
[RSPEC-1751](https://jira.sonarsource.com/browse/RSPEC-1751)
The implementation of rule should not raise issue for continue (#954 will do that instead), and it should raise issue for cases like this:
```javascript
while (condition1) {
   if (condition2) {
      doSomething(); break;
   } else  {
      doSomethingElse(); break;
   }
}
```

**Comment by vilchik-elena** [comment]
fixed by #1494
